From 180cd75c102364d57916ac96944bed1f56a3f447 Mon Sep 17 00:00:00 2001
From: Philip Withnall <philip@tecnocode.co.uk>
Date: Wed, 1 Nov 2023 11:10:00 +0000
Subject: [PATCH 22/24] tests: Fix dependency of test.gresource on
 test-generated.txt

`test-generated.txt` is listed in `test.gresource.xml`, so it needs to
be specified as a dependency in the `custom_target()` which uses
`test.gresource.xml`.

Fixes intermittent build failures like:
```
FAILED: gio/tests/test.gresource
/builds/GNOME/glib/_build/gio/glib-compile-resources --compiler=gcc --target=gio/tests/test.gresource --sourcedir=/builds/GNOME/glib/gio/tests --sourcedir=/builds/GNOME/glib/_build/gio/tests --internal ../gio/tests/test.gresource.xml
../gio/tests/test.gresource.xml: Failed to locate test-generated.txt in any source directory.
```

See !3671 and #3163.

Signed-off-by: Philip Withnall <philip@tecnocode.co.uk>
(cherry picked from commit a61888bceb2b4d23bdc82e579fa3c99c69a7a9fa)
---
 gio/tests/meson.build | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/gio/tests/meson.build b/gio/tests/meson.build
index dbe7c38cc..fdb8f5e22 100644
--- a/gio/tests/meson.build
+++ b/gio/tests/meson.build
@@ -687,8 +687,12 @@ if meson.can_run_host_binaries()
     output : ['gresource-big-test.txt'],
     command : [python, '@INPUT0@', '@OUTPUT@'])
 
+  # referenced by test.gresource.xml
+  test_generated_txt = fs.copyfile('test1.txt', 'test-generated.txt')
+
   test_gresource = custom_target('test.gresource',
     input : 'test.gresource.xml',
+    depends : test_generated_txt,
     output : 'test.gresource',
     command : [glib_compile_resources,
                compiler_type,
@@ -766,9 +770,6 @@ if meson.can_run_host_binaries()
                '--manual-register',
                '@INPUT@'])
 
-  # referenced by test.gresource.xml
-  test_generated_txt = fs.copyfile('test1.txt', 'test-generated.txt')
-
   resources_extra_sources = [
     test_gresource,
     test_resources_c,
-- 
2.44.0.windows.1

